modules/gapi/misc/python/pyopencv_gapi.hpp

```diff
@@ -32,38 +58,71 @@ bool pyopencv_to(PyObject* obj, GRunArgs& value, const ArgInfo& info)
     return pyopencv_to_generic_vec(obj, value, info);
 }
 
-static PyObject* from_grunarg(const GRunArg& v)
+template <>
+PyObject* pyopencv_from(const cv::detail::OpaqueRef& o)
+{
+    switch (o.getKind())
+    {
+        case cv::detail::OpaqueKind::CV_BOOL    : return pyopencv_from(o.rref<bool>());
+        case cv::detail::OpaqueKind::CV_INT     : return pyopencv_from(o.rref<int>());
+        case cv::detail::OpaqueKind::CV_DOUBLE  : return pyopencv_from(o.rref<double>());
+        case cv::detail::OpaqueKind::CV_FLOAT   : return pyopencv_from(o.rref<float>());
+        case cv::detail::OpaqueKind::CV_STRING  : return pyopencv_from(o.rref<std::string>());
+        case cv::detail::OpaqueKind::CV_POINT   : return pyopencv_from(o.rref<cv::Point>());
+        case cv::detail::OpaqueKind::CV_POINT2F : return pyopencv_from(o.rref<cv::Point2f>());
+        case cv::detail::OpaqueKind::CV_SIZE    : return pyopencv_from(o.rref<cv::Size>());
+        case cv::detail::OpaqueKind::CV_RECT    : return pyopencv_from(o.rref<cv::Rect>());
```

Comment:
No, because the macro describes types not related with `OpaqueKind`. For example: GArray in python can have a `GMat` type, but obviously that type doesn't exist in `OpaqueKind` and so on.

Reply:
Don't you mind resolving this one ?